### ISSUE TITLE
Standardize return types for set(...) Methods

### ISSUE THREAD

**Issue by jmarrec** [NREL/OpenStudio#2620]
Using `SizingZone` as an example.

From [SizingZone_Impl.hpp#L298](https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/SizingZone_Impl.hpp#L298)

```cpp
void setZoneHeatingDesignSupplyAirTemperature(double zoneHeatingDesignSupplyAirTemperature);
bool setZoneHeatingDesignSupplyAirTemperature(const Quantity& zoneHeatingDesignSupplyAirTemperature);
```

Is there any reason the first one that accepts a double would be `void` while the other is `bool`?

I mean 2 lines below you do have 

```cpp 
bool setZoneCoolingDesignSupplyAirHumidityRatio(double zoneCoolingDesignSupplyAirHumidityRatio);
```

**I think this needs to be standardized**, methods that set should always return a bool for example...

As an example, this makes this openstudio-standards code return false when everything worked... https://github.com/NREL/openstudio-standards/blob/master/openstudio-standards/lib/openstudio-standards/standards/Standards.ThermalZone.rb#L1217

**Comment by jmarrec** [comment]
I just pushed a commit on my fork where I've made some changes, and put some "TODO" comments in there.

I won't open a PR, the only thing I'm asking is to start a discussion on this. Moving forward, this should really be unified and standardized I think, and I'm willing to help.

**Comment by jmarrec** [comment]
Do you foresee (backwards) compatibility issues if we were to change this?

**Comment by jmarrec** [comment]
There 2417 setXXX methods currently that return void...

**Comment by jmarrec** [comment]
Closed via c03da60164e16ba3917679c4a38ce47fe86b545a
